website/docs/copa-generate.md

```diff
@@ -0,0 +1,254 @@
+
+## Overview
+
+Instead of directly patching and pushing images like `copa patch`, the `generate` command produces a tar stream containing:
+
+- A Dockerfile with instructions to apply the patch
+- A patch layer directory with the security updates
+
+This output can be piped directly to `docker build` or saved as a tar file for later use, giving you full control over the build process with all of Docker's advanced features.
+
+## Key Advantages over `copa patch`
+
+With `copa generate`, you can use any Docker build flag:
+
+```bash
+# Multi-platform builds
+copa generate -i nginx:1.21.6 | docker buildx build --platform linux/amd64,linux/arm64 -t nginx:patched --push -
+
+# Use build cache
+copa generate -i nginx:1.21.6 | docker build --cache-from nginx:cache -t nginx:patched -
+
+# Export to different formats
+copa generate -i nginx:1.21.6 | docker build --output type=tar,dest=patched.tar -
+
+# Set custom build arguments
+copa generate -i nginx:1.21.6 | docker build --build-arg HTTP_PROXY=http://proxy:8080 -t nginx:patched -
+
+# Specify target registry directly
+copa generate -i nginx:1.21.6 | docker build -t myregistry.io/nginx:patched --push -
+```
+
+## Use Cases
+
+The `copa generate` command is ideal for:
+
+- **Docker Build Integration** - Use native Docker build flags and features not available in `copa patch`
+- **Multi-platform Builds** - Leverage `docker buildx` for cross-platform patching
+- **CI/CD Pipelines** - Integrate patches using standard Docker commands your team already knows
+- **Build Cache Optimization** - Use Docker's layer caching mechanisms for faster builds
+- **Air-gapped Environments** - Create patch contexts offline and apply them later
+- **Custom Registries** - Push directly to any registry using Docker's native push capabilities
+
+## Command Syntax
+
+```bash
+copa generate [flags]
+```
+
+### Flags
+
+| Flag | Short | Description | Default |
+|------|-------|-------------|---------|
+| `--image` | `-i` | Application image name and tag to patch **(required)** | |
+| `--report` | `-r` | Vulnerability report file path (optional) | |
+| `--tag` | `-t` | Tag for the patched image | |
+| `--tag-suffix` | | Suffix for the patched image (if no explicit --tag provided) | `patched` |
+| `--output-context` | | Path to save the generated tar context (instead of stdout) | stdout |
+| `--scanner` | `-s` | Scanner that generated the report | `trivy` |
+| `--addr` | `-a` | Address of buildkitd service | local docker daemon |
+| `--working-folder` | `-w` | Working folder | system temp folder |
+| `--timeout` | | Timeout for the operation | `5m` |
+| `--ignore-errors` | | Ignore errors during patching | `false` |
+| `--format` | `-f` | Output format for VEX document | `openvex` |
+| `--output` | `-o` | Output file path for VEX document | |
```

Comment:
can we output vex using copa generate?

Reply:
Yes, we can, I've added an additional e2e test specifically for the vex output which found a bug which has also been solved in the latest commit 👍